websites/R/Radio.net/presence.ts

```diff
@@ -1,187 +1,262 @@
+    // Corporate page
+    switch (path[0]) {
+      // About us
+      // Broadcasters
+      // Advertising
+      // Press
+      // Jobs
+      // Contact
+      case "about-us":
+      case "ueber-uns":
+      case "broadcasters":
+      case "sender":
+      case "advertising":
+      case "werbung":
+      case "press":
+      case "presse":
+      case "jobs":
+      case "contact":
+      case "kontakt":
+        await handleCorporate(host.slice(1).join("."));
+        break;
+      default:
+        presence.setActivity();
+        break;
+    }
+  } else {
+    // Main page
+    switch (path[0]) {
+      // Radio station
+      case "s":
+        await handleStation();
+        break;
+      // Podcast
+      case "p":
+        await handlePodcast();
+        break;
+      // Search
+      case "search":
+        await handleSearch();
+        break;
+      // Genre
+      // Topic
+      // Country
+      // City
+      // Local stations
+      // Top stations
+      case "genre":
+      case "topic":
+      case "country":
+      case "city":
+      case "local-stations":
+      case "top-stations":
+        await handleListing();
+        break;
+      // Choose your country
+      // Contact
+      // App
+      // Terms and conditions
+      // Privacy policy
+      // Imprint
+      case "country-selector":
+      case "contact":
+      case "app":
+      case "terms-and-conditions":
+      case "privacy-policy":
+      case "imprint":
+        await handleGeneric(true);
+        break;
+      // Startpage
+      // Unknown
+      default:
+        presence.setActivity();
+        break;
     }
-    //Search
-    case "search": {
-      browsingTimestamp = 0;
+  }
+});
+
+/**
+ * Get Language Strings
+ * @returns Language Strings
+ */
+async function getStrings(): Promise<LangStrings> {
+  return presence.getStrings(
+    {
+      play: "general.playing",
+      pause: "general.paused",
+      search: "general.searching",
+      browsing: "general.browsing"
+    },
+    (await presence.getSetting("lang").catch(() => "en")) as string
+  );
+}
+
+/**
+ * Handle radio station
+ */
+async function handleStation(): Promise<void> {
+  let presenceData: PresenceData;
+
+  // Check if the animation icon is shown
+  if (
+    document.querySelector<HTMLElement>(".player__animate-icon").style
+      .display !== "none"
+  ) {
+    // Radio is playing / buffering
+    if (timestamp === 0) timestamp = Date.now();
+
+    presenceData = {
+      details: document.querySelector("h1").textContent,
+      state: document.querySelector<HTMLElement>(".player__song").textContent,
+      largeImageKey: (
+        document.querySelector<HTMLElement>("#station").children[3].children[1]
+          .firstChild.firstChild.firstChild as HTMLImageElement
+      ).src,
+      smallImageText: (await strings).play,
+      smallImageKey: "play",
+      startTimestamp: timestamp
+    };
+  } else {
+    // Radio is paused
+    timestamp = 0;
```

Comment:
Not needed as `timestamp` is not used beyond this point?

Reply:
Acts as reset. Otherwise if you would listen to a station and switch to another, the timestamp wouldn't overwrite.